test/precompile.jl

```diff
@@ -797,4 +797,25 @@ let
     end
 end
 
+# Issue #25971
+let
+    load_path = mktempdir()
+    load_cache_path = mktempdir()
+    try
+        pushfirst!(LOAD_PATH, load_path)
+        pushfirst!(DEPOT_PATH, load_cache_path)
+        sourcefile = joinpath(load_path, "Foo25971.jl")
+        write(sourcefile, "module Foo25971 end")
+        chmod(sourcefile, 0o666)
+        cachefile = Base.compilecache(Base.PkgId("Foo25971"))
+        @test filemode(sourcefile) == filemode(cachefile)
+        chmod(sourcefile, 0o600)
+        cachefile = Base.compilecache(Base.PkgId("Foo25971"))
+        @test filemode(sourcefile) == filemode(cachefile)
+    finally
+        rm(load_path, recursive=true)
+        rm(load_cache_path, recursive=true)
```

Comment:
Even though CI is passing, should the updates to `LOAD_PATH` and `DEPOT_PATH` be reversed again?

```suggestion
        rm(load_cache_path, recursive=true)
        filter!(s -> s != load_path, LOAD_PATH)
        filter!(s -> s != load_cache_path, DEPOT_PATH)
```

Reply:
I should suggest using `popfirst!(LOAD_PATH)` instead.

Reply:
I've thought about that, too, but using `filter!` would be more robust in case of a failure in between or even before the `pushfirst!`s, or should the tests be run concurrently some time in the future.

Reply:
Another way is to `copy` before running the code and then `copy!` it back when done.

Reply:
Removed those paths as @jonas-schulze suggested. Thank you.

Reply:
FWIW, `mktempdir` now automatically deletes the folder when the process exits.

Reply:
Still a best practice to delete when done using.